utils/build.ps1

```diff
@@ -3284,6 +3300,37 @@ function Copy-BuildArtifactsToStage([Hashtable] $Platform) {
   Invoke-Program "$($WiX.Path)\wix.exe" -- burn detach "$BinaryCache\$($Platform.Triple)\installer\Release\$($Platform.Architecture.VSName)\installer.exe" -engine "$Stage\installer-engine.exe" -intermediateFolder "$BinaryCache\$($Platform.Triple)\installer\$($Platform.Architecture.VSName)\"
 }
 
+function Build-NoAssertsToolchain() {
+  if ($ToBatch) {
+    Write-Output ""
+    Write-Output "Building NoAsserts Toolchain ..."
+  } else {
+    Write-Host -ForegroundColor Cyan "[$([DateTime]::Now.ToString("yyyy-MM-dd HH:mm:ss"))] Building NoAsserts Toolchain ..."
+  }
+  $Stopwatch = [Diagnostics.Stopwatch]::StartNew()
+
+  Invoke-BuildStep Build-Compilers $HostPlatform -Variant "NoAsserts"
```

Comment:
How does this trigger a build without no assertions, I don't see that in the change.

Reply:
I know that the function is being called, but the call will still build an asserts enabled toolchain.

Reply:
Yes. We need to build an assert layout to copy the binaries from. So we first build the whole toolchain as usual with asserts, then we call this function. It builds the no assert compilers and then copies over alll the remaining files to build a full layout

Reply:
I don't see how though. It looks like it just copies the toolchain and then builds another *asserts* toolchain and installs that into it. There is no control over `LLVM_ENABLE_ASSERTS`.

Reply:
Ah, there is already a check for `$Variant -eq "Asserts"`.